docs/architecture/ADR001.md

```diff
@@ -0,0 +1,25 @@
+# ADR001: Include record for services/resources provisioned for tenants
+
+## Status
+
+Accepted
+
+## Context
+
+The Paas billing system receives a series of events from Cloudfoundry notifying whether services/resources have been created/renamed/deleted, when, for which tenant, etc.
+
+The original Paas billing system then goes through several stages of processing, translating the Cloudfoundry events into records of services/resources by calendar month before calculating the final monthly bill for each tenant. This complex process is done every month, at the start of the month, and there is no persistent record of the results of each stage of processing, including what services/resources tenants have provisioned. After each stage of processing database tables are populated but the contents of these tables are impermanent, being refreshed each month.
+
+If there is a problem with a tenant's bill it is very difficult to go backwards from the bill to find the source of the problem.
+
+## Decision
+
+There is no need to regenerate historical records of services/resources provisioned for tenants each time billing is run each month since this information does not change. Furthermore, recording this information for each month makes it difficult for us to calculate bills between any two dates and times. We instead need to have a persistent record of the services/resources each tenant has provisioned with a from- and to-date denoting when the service/resource was/is being used. This persistent record is in the new `resources` table.
```

Comment:
I'm not sure I understand why historical records are mentioned here - is this something the old billing did? If so, I think it should be under "context". 
The "Decision" itself starts from "We need a persistent record..."

Reply:
```suggestion
There is no need to regenerate historical records of services/resources provisioned for tenants each time billing is run each month since this information does not change. Furthermore, recording this information for each month makes it difficult for us to calculate bills between any two dates and times. We instead need to have a persistent record of the services/resources each tenant has provisioned with `from` and `to` dates denoting when the service/resource was/is being used. This persistent record is in the new `resources` table.
```